scipy/interpolate/_pade.py

```diff
@@ -41,19 +44,23 @@ def pade(an, m):
 
     """
     an = asarray(an)
-    N = len(an) - 1
-    n = N - m
-    if n < 0:
+    if l is None:
+        l = len(an) - 1 - m
+    N = m + l
+    if l < 0:
         raise ValueError("Order of q <m> must be smaller than len(an)-1.")
-    Akj = eye(N+1, n+1)
+    if N > len(an)-1:
+        raise ValueError("Order of q+p <m+l> must be smaller than len(an).")
```

Comment:
These error messages can maybe be improved. The first one is no longer accurate.

Reply:
Thank you, a good point. 
I left the initial error message in place for when only <m> is given. If both parameters are given the first error message is never called. If both parameters are given, we only care about the sum of them, hence why only warning the user about the sum.
If you think it is worthwhile, I could how much they are over by to the error message?

Reply:
It'd be nice if you specified the `l < 0` part, what if the user themselves passes in `l < 0`?